lib/senders/email.js

```diff
@@ -298,6 +298,10 @@ module.exports = function (log, config) {
       headers: Object.keys(headers).join(',')
     })
 
+    if (config.emailService.forcedEmailAddresses.test(emailConfig.to)) {
+        this.mailer = require('./email_service')(config)
```

Comment:
Do we need to construct a fresh one of these every time we want to call it, or can we just initialise it once at the top of the module and use the same copy every time? It looks like `email_service` contains no state so should be fine to re-use/re-enter?

Reply:
I tried to initialize it along with the rest of the stuff (specifically here: https://github.com/mozilla/fxa-auth-server/blob/71dd839d357207eb9174bf4b4cd21df84c0ecd88/lib/senders/email.js#L138), but the `config` there is different and doesnt have `emailService`. Is there a better way around it though? I also don't like my solution.

Reply:
Oh right, yeah, I see now.

I think you need to set it in the `Mailer` constructor in that case, call it something like `this.externalMailer` or whatever. Then rather than clobbering `this.mailer` here, have a local `mailer` variable that is either set to `this.mailer` or `this.externalMailer` depending on the email address.

Does that make sense? Or am I still misunderstanding how it works?

Reply:
Makes sense, yeah.

Reply:
...oh, *now* I understand what you were saying, sorry!

So earlier, when I said "initialise it once at the top of the module" I was really thinking further up, next to `oauthClientInfo` on line 25. After that, I only suggested the `Mailer` constructor because I mistakenly thought you meant it needed that config. But seeing your test case, it's better this way because it means you can stub it. So :+1:. 

And the other part of this change, renaming to `mailerConfig` to avoid the naming clash, makes sense regardless of that anyway fwiw. Shadowing the outer name was demonstrably confusing, given how confused I was until just now, so let's definitely keep that!